test/modules/EdgeHubRestartTester/src/DirectMethodEdgeHubConnectorTest.cs

```diff
@@ -15,95 +15,82 @@ namespace EdgeHubRestartTester
-        readonly DateTime edgeHubRestartedTime;
-        readonly uint restartSequenceNumber;
         readonly ILogger logger;
         long directMethodCount = 0;
+        ModuleClient dmModuleClient = null;
+        TestResultReportingClient reportClient = null;
 
         public DirectMethodEdgeHubConnectorTest(
-            ModuleClient dmModuleClient,
             Guid batchId,
-            DateTime runExpirationTime,
-            DateTime edgeHubRestartedTime,
-            uint restartSequenceNumber,
-            ILogger logger,
-            CancellationToken cancellationToken)
+            ILogger logger)
         {
-            this.dmModuleClient = Preconditions.CheckNotNull(dmModuleClient, nameof(dmModuleClient));
             this.batchId = batchId;
-            this.runExpirationTime = runExpirationTime;
-            this.cancellationToken = Preconditions.CheckNotNull(cancellationToken, nameof(cancellationToken));
-            this.edgeHubRestartedTime = edgeHubRestartedTime;
-            this.restartSequenceNumber = restartSequenceNumber;
             this.logger = Preconditions.CheckNotNull(logger, nameof(logger));
         }
 
-        public async Task StartAsync()
+        public void Dispose() => this.dmModuleClient?.Dispose();
+
+        public async Task StartAsync(
+            DateTime runExpirationTime,
+            DateTime edgeHubRestartedTime,
+            CancellationToken cancellationToken)
         {
             (DateTime dmCompletedTime, HttpStatusCode dmStatusCode) = await this.SendDirectMethodAsync(
                 Settings.Current.DeviceId,
                 Settings.Current.DirectMethodTargetModuleId,
-                this.dmModuleClient,
                 Settings.Current.DirectMethodName,
-                this.runExpirationTime,
-                this.cancellationToken,
-                this.logger);
+                runExpirationTime,
+                cancellationToken);
 
             TestResultBase dmTestResult = new EdgeHubRestartDirectMethodResult(
-                Settings.Current.ModuleId + "." + TestOperationResultType.EdgeHubRestartDirectMethod.ToString(),
+                this.GetSource(),
                 DateTime.UtcNow,
                 Settings.Current.TrackingId,
                 this.batchId,
-                Interlocked.Read(ref this.directMethodCount).ToString(),
-                this.edgeHubRestartedTime,
+                this.directMethodCount.ToString(),
+                edgeHubRestartedTime,
                 dmCompletedTime,
-                dmStatusCode,
-                this.restartSequenceNumber);
+                dmStatusCode);
 
-            var reportClient = new TestResultReportingClient { BaseUrl = Settings.Current.ReportingEndpointUrl.AbsoluteUri };
-            await ModuleUtil.ReportTestResultUntilSuccessAsync(
-                reportClient,
+            await ModuleUtil.ReportTestResultAsync(
+                this.GetReportClient(),
```

Comment:
remove this.GetReportClient() and this.logger from this method call.

Reply:
`ReportTestResultAsync()` lives in `ModuleUtil` class which shared across multiple classes, not only within `IEdgeHubConnectorTest`